### PR TITLE
add sets support

### PR DESCRIPTION
**Pull request by Lakate**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,7 +19,10 @@ module.exports = function(source) {
 
     const callback = this.async(),
         options = Object.assign({}, this.options.bemLoader, loaderUtils.getOptions(this)),
-        levelsMap = options.levels || bemConfig.levelMapSync(),
+        setLevels = options.sets && bemConfig.levelsSync(options.sets) || [],
```
Comment (Yeti-or) [inline index.js]: Could you describe desired behaviour
and add it to README section with options

File: index.js [hunk 2]
```diff
@@ -19,7 +19,10 @@ module.exports = function(source) {
...
+        levelsMap = setLevels.length > 0 ?
+            setLevels.map(level => level.path) :
+            (options.levels || bemConfig.levelMapSync()),
```
Comment (Vittly) [inline index.js]: I can write shorter version:

```js
levelsFromSets = options.sets && bemConfig.levelsSync(options.sets).map(level => level.path),
levelsMap = levelsFromSets || options.levels || bemConfig.levelMapSync()
```

File: package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "@bem/fs-scheme": "^2.1.0",
     "@bem/import-notation": "^1.1.2",
     "@bem/naming": "^2.0.0-6",
-    "bem-config": "^3.2.3",
+    "@bem/sdk.config": "0.0.3",
```
Comment (Vittly) [inline package.json]: very-very young version)

Comment (Vittly) [reply package.json]: we also should update version of loader

File: index.js (hunk 1 above)
Comment (Vittly) [reply index.js]: I think we should add examples inside `sdk.config` and refer to them

Comment (Lakate) [reply index.js]: in my opinion, we should add describe to README, and also add example inside sdk.config

File: README.md [hunk 4]
```diff
@@ -75,10 +75,51 @@ module: {
...
+    rules: [
+        {
+            test : /\react.js$/,
+            loader: 'webpack-bem-loader',
+            options: {
+                naming: 'react',
+                levels: {
+                    './pathToLevel1Blocks': {
+                        default: true,
+                        scheme: 'nested',
```
Comment (Lakate) [inline README.md]: убрать

File: index.js [hunk 5]
```diff
@@ -19,7 +19,8 @@ module.exports = function(source) {
 
     const callback = this.async(),
         options = Object.assign({}, this.options.bemLoader, loaderUtils.getOptions(this)),
-        levelsMap = options.levels || bemConfig.levelMapSync(),
+        levelsFromSets = options.sets && bemConfig.levelsSync(options.sets).map(level => level.path),
+        levelsMap = levelsFromSets || options.levels || bemConfig.levelMapSync()
```
Comment (Lakate) [inline index.js]: передавать только set

File: README.md [hunk 6]
```diff
@@ -79,6 +79,7 @@ module: {
 
 - __naming__: [bem-naming](https://en.bem.info/toolbox/sdk/bem-naming) overrides
 - __levels__ <Array>: paths to components declarations
+- __set__ <String>: string with name of set
```
Comment (Yeti-or) [inline README.md]: and It be cool to have link to this docs: https://github.com/bem/bem-sdk/pull/213/files#diff-621cc23937ce7f99fed2fab23dbbbee0R160
but not necessary right now

**Comment by Lakate** [comment]
i'l fix after merge https://github.com/bem/bem-sdk/pull/213

File: package.json [hunk 7]
```diff
@@ -20,7 +20,7 @@
     "@bem/fs-scheme": "^2.1.0",
     "@bem/import-notation": "^1.1.2",
     "@bem/naming": "^2.0.0-6",
-    "bem-config": "^3.2.3",
+    "@bem/sdk.config": "0.1.0-alpha.2bee4230",
```
[the remark this answers is not in the document]
Comment (Lakate) [reply package.json]: use 0.0.4

**Comment by Yeti-or** [comment]
@Lakate `@bem/sdk.config@0.0.4` is out

File: README.md [hunk 8]
```diff
@@ -79,6 +79,7 @@ module: {
 
 - __naming__: [bem-naming](https://en.bem.info/toolbox/sdk/bem-naming) overrides
 - __levels__ <Array>: paths to components declarations
+- __set__ <String>: string with name of set. See more on https://github.com/bem/bem-sdk/blob/master/packages/config/README.md
```
Comment (Vittly) [inline README.md]: simply "name of set"

Comment (Vittly) [reply README.md]: can we add direct link to "set definition section"? with `...#section-foo`

File: index.js [hunk 9]
```diff
@@ -19,7 +19,8 @@ module.exports = function(source) {
 
     const callback = this.async(),
         options = Object.assign({}, this.options.bemLoader, loaderUtils.getOptions(this)),
-        levelsMap = options.levels || bemConfig.levelMapSync(),
+        levelsFromSets = options.sets && bemConfig.levelsSync(options.sets).map(level => level.path),
+        levelsMap = levelsFromSets || bemConfig.levelMapSync() || options.levels,
```
Comment (Vittly) [inline index.js]: Hmm. Why do we use `levelsSync().map()` above? Is `levelMapSync` doing same thing?

Comment (Vittly) [reply index.js]: This is all about strange method names... ok

**Comment by Lakate** [comment]
@Yeti-or merge? roll out a new version ?

**Comment by Lakate** [comment]
желательно обновить до @0.4.2, чтобы подтянулось в ```bem-react-scripts```

**Comment by awinogradov** [comment]
It's not needed here. It's supported in https://github.com/bem/create-bem-react-app/pull/31 and  https://github.com/bem/create-bem-react-app/pull/32.

**Comment by Yeti-or** [comment]
@awinogradov Why it's not needed here??

**Comment by awinogradov** [comment]
Because it's loader. Loader works with targets from webpack-config. Also, we don't need sets infromation in the loader, because of no reasons. Also, we have instruments to do this support early and parallel. Look at the comment above. Also, supporting sets here can't help to do anything.

**Comment by awinogradov** [comment]
@Lakate @Vittly is it actual PR or you will support this in https://github.com/bem/webpack-bem-plugin?

**Comment by Vittly** [comment]
It is already supported in plugin. Can be closed

**Pull request closed without merging**